multicluster/cmd/multicluster-controller/memberclusterannounce_webhook.go

```diff
@@ -67,20 +74,20 @@ func (v *memberClusterAnnounceValidator) Handle(ctx context.Context, req admissi
 	}
 
 	clusterSet := clusterSetList.Items[0]
-	if clusterSet.Name == memberClusterAnnounce.ClusterSetID {
-		for _, member := range clusterSet.Spec.Members {
-			if member.ClusterID == memberClusterAnnounce.ClusterID {
-				// validate the ServiceAccount used is correct
-				if member.ServiceAccount == saName {
-					return admission.Allowed("")
-				} else {
-					return admission.Denied("Member does not have permissions")
-				}
-			}
-		}
+	if len(clusterSet.Spec.Leaders) != 1 {
+		return admission.Errored(http.StatusPreconditionFailed,
+			fmt.Errorf("invalid ClusterSet config in the leader cluster, there must be one leader cluster in the ClusterSet"))
 	}
 
-	return admission.Denied("Unknown member")
+	if clusterSet.Name == memberClusterAnnounce.ClusterSetID &&
+		clusterSet.Spec.Leaders[0].ClusterID == memberClusterAnnounce.LeaderClusterID {
+		return admission.Allowed("")
+	}
+
+	if clusterSet.Name != memberClusterAnnounce.ClusterSetID {
+		return admission.Denied("Unknown ClusterSet ID")
+	}
+	return admission.Denied("The leader cluster ID in the MemberClusterAnnounce is not defined in the leader ClusterSet config")
```

Comment:
"Leader cluster ID in the MemberClusterAnnounce does not match that in the ClusterSet"

Reply:
About the length check of the ClusterSet.Spec.Leader, maxItem and minItem will help. So I will use that way to avoid checking the length